examples/with-nextjs-and-clerk-auth/src/lib/monite-api/demo-data-generator/counterparts.ts

```diff
@@ -262,26 +265,36 @@ export const createCounterpart = async ({
 
 export const createCounterpartVatId = async ({
   counterpart_id,
-  entity_id,
+  entity,
   token,
 }: {
   counterpart_id: string;
-  entity_id: string;
+  entity: components['schemas']['EntityOrganizationResponse'];
   token: AccessToken;
 }): Promise<CounterpartVatIDResponse> => {
-  const { POST } = createMoniteClient({
-    headers: {
-      Authorization: `${token.token_type} ${token.access_token}`,
-    },
-  });
+  const { POST } = createMoniteClient(token);
 
-  const type = getRandomItemFromArray([
-    'eu_vat',
-    'no_vat',
-    'unknown',
-  ] satisfies Array<VatIDTypeEnum>);
   const value = String(faker.number.int(10_000));
   const addressCountries = ['DE', 'US', 'GB'] satisfies Array<AllowedCountries>;
+  const counterpartCountry = getRandomItemFromArray(addressCountries);
+  const entityCountry = entity.address.country;
+  let vatIdType: VatIDTypeEnum = 'unknown';
```

Comment:
I would caution you against using `let` in functions with a large number of other variables. It is rather difficult to keep track of where `let` changes.
In my opinion, if you put this logic in a separate function and solve everything with pure `if`, it will be easier to read.